pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -240,7 +240,13 @@ func (c *Reconciler) makeBuild(tr *v1alpha1.TaskRun, logger *zap.SugaredLogger)
 	}
 
 	// Apply parameters from the taskrun.
-	build = applyParameters(b, t, tr)
+	build = applyParameters(build, t, tr)
+
+	// Apply resources from the taskrun.
+	build, err = applyResources(build, t, tr, c.resourceLister)
```

Comment:
we make a call to `addInputResources` just a few lines before that, which loops on all the resources to add them to the build type. Maybe we can do the apply parameters there as well to be more efficient

Reply:
Hmm, I think that would make the code harder to read and test. The lister should be cached anyway, so I'm not sure it would really improve performance. I could do a single loop to get all the resources then pass them into these functions, but this seems simpler and about the same performance.